integration/e2e/limits_test.go

```diff
@@ -90,6 +98,104 @@ func TestLimits(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestOTLPLimits(t *testing.T) {
+	s, err := e2e.NewScenario("tempo_e2e")
+	require.NoError(t, err)
+	defer s.Close()
+
+	require.NoError(t, util.CopyFileToSharedDir(s, configLimits, "config.yaml"))
+	tempo := util.NewTempoAllInOne()
+	require.NoError(t, s.StartAndWaitReady(tempo))
+
+	protoSpans := test.MakeProtoSpans(100)
+
+	// gRPC
+	grpcClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithEndpoint(tempo.Endpoint(4317)),
+		otlptracegrpc.WithInsecure(),
+		otlptracegrpc.WithRetry(otlptracegrpc.RetryConfig{Enabled: false}),
+	)
+	require.NoError(t, grpcClient.Start(context.Background()))
+
+	grpcErr := grpcClient.UploadTraces(context.Background(), protoSpans)
+	assert.Error(t, grpcErr)
+	require.Equal(t, codes.ResourceExhausted, status.Code(grpcErr))
+
+	// HTTP
+	httpClient := otlptracehttp.NewClient(
+		otlptracehttp.WithEndpoint(tempo.Endpoint(4318)),
+		otlptracehttp.WithInsecure(),
+		otlptracehttp.WithRetry(otlptracehttp.RetryConfig{Enabled: false}),
+	)
+	require.NoError(t, httpClient.Start(context.Background()))
+
+	httpErr := httpClient.UploadTraces(context.Background(), protoSpans)
+	assert.Error(t, httpErr)
+	require.Contains(t, httpErr.Error(), "retry-able request failure")
```

Comment:
Would this fail on `main` using the non-forked receiver?

Reply:
Okay I see now, thanks for pointing that out.

Reply:
It looks like the PR upstream has been merged.  Should we make the adjustment to this PR also?